clients/include/rocblas_test.hpp

```diff
@@ -54,7 +54,19 @@
         }                                         \
     } while(0)
 
-#define EXPECT_ROCBLAS_STATUS ASSERT_EQ
+// This wraps the rocBLAS call with catch_signals_and_exceptions_as_failures().
+// By placing it at the rocBLAS call site, memory resources are less likely to
+// be leaked in the event of a caught signal.
+#define EXPECT_ROCBLAS_STATUS(STATUS, EXPECT)                               \
+    do                                                                      \
```

Comment:
CHECK_ROCBLAS_ERROR is defined to this too, which is probably alright.   We seem to have some inconsistent checks to status and error in some of the benchmark timing loops which we will likely want to remove.

Reply:
> 
> 
> CHECK_ROCBLAS_ERROR is defined to this too, which is probably alright. We seem to have some inconsistent checks to status and error in some of the benchmark timing loops which we will likely want to remove.

`#ifdef GOOGLE_TEST` is used to tell `rocblas-test` apart from `rocblas-bench`. The same macros are used by both, but they have different definitions for `rocblas-test` and `rocblas-bench`. This change only affects `rocblas-test`, and so it only modifies the macros which are defined when `GOOGLE_TEST` is defined. We do not want them to be consistent except maybe in their API, because they have different requirements between unit testing and benchmarking.